includes/edit.form.inc

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/**
+ * @file
+ * Form functions for editing a MODS datastream by re-transforming MARCXML.
+ */
+
+/**
+ * Form to allow editing of an existing MODS datastream.
+ *
+ * @param array $form
+ *   The Drupal form.
+ * @param array $form_state
+ *   The Drupal form state.
+ * @param AbstractObject $object
+ *   A object within Fedora.
+ *
+ * @return array
+ *   A Drupal form definition.
+ */
+function islandora_marcxml_edit_file_form(array $form, array &$form_state, AbstractObject $object) {
+  $form = array();
+  $form['file'] = array(
+    '#type' => 'managed_file',
+    '#title' => t('MARCXML File'),
+    '#description' => t('A file containing a MARCXML record, to be transformed to MODS.'),
+    '#upload_validators' => array(
+      'file_validate_extensions' => array('xml'),
+    ),
+    '#required' => TRUE,
+  );
+  $form['submit'] = array(
+    '#type' => 'submit',
+    '#value' => t('Update'),
+  );
+  return $form;
+}
+
+/**
+ * Populate the MODS datastream with the transformed MARCXML.
+ *
+ * @param array $form
+ *   The Drupal form.
+ * @param array $form_state
+ *   The Drupal form state.
+ */
+function islandora_marcxml_edit_file_form_submit(array &$form, array &$form_state) {
+  module_load_include('inc', 'islandora_marcxml', 'includes/utilities');
+  $file = file_load($form_state['values']['file']);
+  $object = islandora_object_load($form_state['build_info']['args'][0]->id);
+  islandora_marcxml_populate_mods($object, $file);
```

Comment:
Hey, still on my radar and a few hours still left for the 24 hrs mark. But i have a questions: this is not the normal edit function because it involves a derivative process. Which means uploaded(and then deleted file) gets converted into MODS. A) If the transformation fails (empty result) we should invalidate this somehow and return without making changes. B)Since this is doing a transformation, which is new to what we were doing previously, but does not act as normally a Form does (because it skips the MODS to DC!) we should document this somehow, or, act as for editing would do when editing MODS via XML form. Generate also a new DC? But not sure how... DC would end in an undefined state here.